### PR TITLE
Fix catalog for flink 1.12

### PR DESCRIPTION
**Pull request by jianyun8023**

### CODE REVIEW

**Comment by jianyun8023** [comment]
@reswqa There are still two tests in CatalogITest (CatalogITest#testTableSink, CatalogITest#testSinkToExistingTopic) that have not been fixed, please help me fix them. They are caused by the same cause.

File: pulsar-flink-connector/src/main/java/org/apache/flink/table/catalog/pulsar/factories/PulsarCatalogFactory.java [hunk 1]
```diff
@@ -66,6 +66,8 @@ public Catalog createCatalog(String name, Map<String, String> properties) {
         props.add(CATALOG_ADMIN_URL);
         props.add(CATALOG_STARTUP_MODE);
         props.add(CATALOG_DEFAULT_PARTITIONS);
+        props.add(org.apache.flink.streaming.connectors.pulsar.table.PulsarOptions.SCAN_STARTUP_MODE.key());
+        props.add(FormatDescriptorValidator.FORMAT);
```
Comment (reswqa) [inline pulsar-flink-connector/src/main/java/org/apache/flink/table/catalog/pulsar/factories/PulsarCatalogFactory.java:70]: Do we need add support for those configurations of key and value like key.xxx

**Review by reswqa — COMMENTED** [review]
LGTM

File: pulsar-flink-connector/src/main/java/org/apache/flink/table/catalog/pulsar/factories/PulsarCatalogFactory.java (hunk 1 above)
Comment (jianyun8023) [reply pulsar-flink-connector/src/main/java/org/apache/flink/table/catalog/pulsar/factories/PulsarCatalogFactory.java:70]: Yes, support needs to be added.

**Review by reswqa — APPROVED** [review]

**Pull request merged**